### ISSUE TITLE
new map signature is required

### ISSUE THREAD

**Issue by maksimlikharev** [modelmapper/modelmapper#132]
```
public <S, D> D map(S source, Class<S> sourceType, Type destinationType, String typeMapName) 
```

as it seems like the only way to map Interface to Object and have all your converters and Co working, otherwise it will not find configuration

**Comment by jhalterman** [comment]
The `<D> D map(Object source, Type destinationType, String typeMapName)` method doesn't work? What problem are you seeing specifically?

**Comment by maksimlikharev** [comment]
Here is the problem I have:

1. interface User { }
2. class UserDTO implements User { } <== this guy deep inside the dao and not visible
3. class UserEntity

I need to map User ==> UserEntity

I define mapping as:
			mapper.addMappings( new PropertyMap<User, UserEntity>() {

and all my converters fall off, as it really want to produce mapping as  UserDTO  ==> UserEntity and cannot fined maptype, basically using default conversion.

Spent 2 hours yesterday hitting my head on the wall, ended up creating my own ModelMapper and producing necessary signatures.

another point, one can reasonably expect that I want to map following:

class Agg implements A, B, C, D
     and map C ==> CPojo

**Comment by chhsiao90** [comment]
Closed by #209

**Comment by chhsiao90** [comment]
The TypeMap inheritance had been release on ModelMapper v1.0.0, please give it a try!
